### PR TITLE
Fixed UniteWithInputDirectory with relative or ~ paths (v2)

### LINKED ISSUE #925: `UniteWithInputDirectory files_rec/async` only works with absolute paths.
It doesn't work with `.` or `../foo` or `~/bar` paths.

### PR DESCRIPTION
**Pull request by louispan**
This is the same as pull request #926 (just with a different head branch), and resolves #925 and #927.

### CODE REVIEW

File: autoload/unite/sources/rec.vim [hunk 1]
```diff
@@ -35,9 +35,10 @@ call unite#util#set_default(
       \ 'g:unite_source_file_rec_max_cache_files')
 call unite#util#set_default('g:unite_source_rec_unit',
       \ unite#util#is_windows() ? 1000 : 2000)
+" -L follows symbolic links to have the same behaviour as file_rec
 call unite#util#set_default(
       \ 'g:unite_source_rec_async_command', (
-      \  !unite#util#is_windows() && executable('find') ? 'find' : ''),
+      \  !unite#util#is_windows() && executable('find') ? 'find -L' : ''),
```
Comment (Shougo) [inline autoload/unite/sources/rec.vim]: Please change the default value in unite.txt.

**Comment by Shougo** [comment]
Thanks.  The documentation problem should be fixed.

**Comment by louispan** [comment]
I've added the documentation for #927.

**Comment by Shougo** [comment]
OK.  Please fix the merge conflict.

**Comment by louispan** [comment]
Merge conflict fixed

**Comment by Shougo** [comment]
Thanks

**Pull request merged**